targets/PythonSdk/source/source/PlayFabHTTP.py

```diff
@@ -0,0 +1,68 @@
+import json
+import PlayFabSettings
+import PlayFabErrors
+import requests
+
+# Note this is a blocking call and will always run synchronously
+# the return type is a dictionary that should contain a valid dictionary that
+# should reflect the expected JSON response
+# if the call fails, there will be a returned PlayFabError
+def DoPost(urlPath, request, authKey, authVal, extraHeaders, callback):
+    url = PlayFabSettings.GetURL(urlPath)
+
+    try:
+        j = json.dumps(request)
+    except e:
+        raise PlayFabErrors.PlayFabException("The given request is not json serializable. {}".format(e))
+
+    requestHeaders = {}
+
+    if extraHeaders:
+        requestHeaders.update(extraHeaders)
+
+    requestHeaders["Content-Type"] = "application/json"
+    requestHeaders["X-PlayFabSDK"] = PlayFabSettings._internalSettings.SdkVersionString
+    requestHeaders["X-ReportErrorAsSuccess"] = "true" # Makes processing PlayFab errors a little easier
+
+    if authKey:
+        requestHeaders[authKey] = authVal
+
+    httpResponse = requests.post(url, data=j, headers=requestHeaders)
+
+    error = response = None
+
+    if httpResponse.status_code != 200:
+        # Failed to contact PlayFab Case
+        error = PlayFabErrors.PlayFabError()
+        error.code = response.status_code
```

Comment:
How easy/difficult would it be to populate error.HttpStatus with the string version of this code?
If it's easy, lets do it. If it's hard, we can wait for a future fix.
Add a todo comment if it's hard.

Reply:
Oh weird...
The parameters we put here don't match the case of the parameters defined in PlayFabErrors.py

Reply:
We should just use the default definition of PlayFabError, and not define every parameter here. Those defaults mostly match what we're creating here, minus the response.status_code which might actually be different.